numpy/core/src/multiarray/arraytypes.c.src

```diff
@@ -328,6 +329,17 @@ string_to_long_double(PyObject*op)
     npy_longdouble temp;
     PyObject* b;
 
+    /* Convert python long objects to a longdouble, without precision or range
+     * loss via a double.
+     */
+    if (PyLong_Check(op)
+#if !defined(NPY_PY3K)
+        || PyInt_Check(op)
+#endif
```

Comment:
I think we just need to avoid running this code-path for booleans (and adding a test for `np.longdouble(True)`)

Reply:
yes! that worked! I guess because of bools in masked arrays or something like that!